modules/container/claws/container/vect.hpp

```diff
@@ -1,241 +1,364 @@
-    constexpr vect<dim, T> map_impl(FUNC &&func, std::index_sequence<indices...>) const
+    template<typename U>
+    constexpr bool operator!=(vect<U, Size> const &other) const noexcept
     {
-      return {func(data[indices])...};
+      return !(*this == other);
     }
 
+  private:
+    template<typename U, typename Pred, size_t... indexes>
+    constexpr auto compare_impl(vect<U, Size> const &other, Pred &&pred, std::index_sequence<indexes...>) const noexcept
+    {
+      return vect<bool, Size>{pred((*this)[indexes], other[indexes])...};
+    };
+
   public:
-    template<class FUNC>
-    constexpr vect<dim, T> map(FUNC &&func) const
-    {
-      return (map_impl(func, std::make_index_sequence<dim>{}));
-    }
-
-#define CLAWS_VECT_UNARY_OP_DEF(OP)              \
-  constexpr vect<dim, T> operator OP(void) const \
-  {                                              \
-    struct mapper                                \
-    {                                            \
-      T operator()(T const &t)                   \
-      {                                          \
-        return OP t;                             \
-      }                                          \
-    };                                           \
-    return map(mapper{});                        \
+    template<typename U, typename Pred>
+    constexpr vect<bool, Size> compare(vect<U, Size> const &other, Pred &&pred) const noexcept
+    {
+      return compare_impl(other, std::forward<Pred>(pred), std::make_index_sequence<Size>{});
+    }
+
+#define CLAWS_VECT_ORDER_COMPARATOR_DEF(OP, NAME)                                         \
+  template<typename U>                                                                    \
+  constexpr vect<bool, Size> is_##NAME(vect<U, Size> const &other) const noexcept         \
+  {                                                                                       \
+    return compare(other, [](const auto &a, const auto &b) constexpr { return a OP b; }); \
+  }
+
+    CLAWS_VECT_ORDER_COMPARATOR_DEF(<, less);
+
+    CLAWS_VECT_ORDER_COMPARATOR_DEF(<=, less_or_equal);
+
+    CLAWS_VECT_ORDER_COMPARATOR_DEF(>, greater);
+
+    CLAWS_VECT_ORDER_COMPARATOR_DEF(>=, greater_or_equal);
+
+#undef CLAWS_VECT_ORDER_COMPARATOR_DEF
+
+#define CLAWS_VECT_UNARY_OP_DEF(OP)                                       \
+  constexpr vect<T, Size> operator OP() const                             \
+  {                                                                       \
+    return vect_transform(*this, [](auto const &cur) { return OP cur; }); \
   }
 
     CLAWS_VECT_UNARY_OP_DEF(-);
+
     CLAWS_VECT_UNARY_OP_DEF(+);
+
     CLAWS_VECT_UNARY_OP_DEF(~);
+
     CLAWS_VECT_UNARY_OP_DEF(!);
 
 #undef CLAWS_VECT_UNARY_OP_DEF
 
-  public:
-    constexpr T sum(void) const
+#define CLAWS_VECT_NAMED_COMPONENT(NAME, INDEX)                           \
+  template<size_t _Size = Size, typename = std::enable_if_t<(_Size > 0)>> \
```

Comment:
the condition checks if Size is greater than 0. It would be better to check that Size > INDEX